plugins/star-rating/api/api.js

```diff
@@ -10,30 +10,349 @@ var plugin = {},
+                            "contact_me":currEvent.segmentation.contactMe,
+                            "rating":currEvent.segmentation.rating,
+                            "widget_id":currEvent.segmentation.widget_id
+                        }, function(err, saved) {
+                            /**
+                             *  register for process new  rating event data.
+                             *  the original event format like:
+                             *  { key: '[CLY]_star_rating', count:1, sum:1, segmentation:{ platform:"iOS", version:"3.2", rating:2}
+                             *  this function will add a field call "platform_version_rate" in segmentation.
+                             */
+                            currEvent.segmentation['platform_version_rate'] = currEvent.segmentation['platform'] + "**" + currEvent.segmentation['app_version'] + "**" + currEvent.segmentation['rating'] + "**";
+                            // remove data from core data
+                            // for GDPR compatibility
+                            delete currEvent.segmentation.email;
+                            delete currEvent.segmentation.comment;       
+                            delete currEvent.segmentation.contactMe;   
+                            delete currEvent.segmentation.widget_id;     
+                            return true;
+                        })
+                    return true;
+                }
+            })
+        }
+    });
+    
+    /*
+    * @apiName: CreateFeedbackWidget
+    * @type: GET
+    * @apiDescription: Create web feedback widget from Countly web application
+    * @apiParam: 'popup_header_text', Header text of feedback popup
+    * @apiParam: 'popup_email_callout', "Contact me by e-mail" text of 
+    * feedback popup
+    * @apiParam: 'popup_comment_callout', "Add comment" text of feedback popup
+    * @apiParam: 'popup_thanks_message', Message of thanks popup
+    * @apiParam: 'trigger_position', position of feedback trigger sticky, 
+    * should be one of these ['mleft','mright','bleft','bright']
+    * @apiParam: 'trigger_bg_color', #hex code of background color of feedback 
+    * trigger sticky button
+    * @apiParam: 'trigger_font_color', #hex code of font color of feedback 
+    * trigger sticky button
+    * @apiParam: 'trigger_button_text', text of feedback sticky button
+    * @apiParam: 'target_devices', target device array of feedback 
+    * fe: ['mobile','tablet']
+    * @apiParam: 'target_page', target page of feedback, should be one of 
+    * these values ['all','selected']
+    * @apiParam: 'target_pages', if 'target_page' property set as 'selected', 
+    * this param should be provided as array of selected pages 
+    * fe: ['/home','/login']
+    * @apiParam: 'is_active', is that feedback should set active as default?
+    * @apiParam: 'app_key', app_key of related application
+    */
+    plugins.register("/i/web-feedback/widgets/create", createFeedbackWidget);
```

Comment:
I don't see any validation for these or other methods. Please user rights.js (https://github.com/Countly/countly-server/blob/master/api/utils/rights.js) to validate if current user calling api has needed rights (read, write, global admin)